### PR TITLE
[PortMidi] Add recipe for v2.0.4

### PR DESCRIPTION
**Pull request by SteffenPL**
Add support for PortMidi.

(This is my first binary builder RP. I hope it works out. Sorry if I break something...)

### CODE REVIEW

File: P/PortMidi/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "PortMidi"
+version = v"0.0.1"
+
+# Collection of sources required to complete build
+sources = [
+    GitSource("https://github.com/PortMidi/portmidi.git", "928520f7b79f371854387cb480b4e3b5bf5dac95")
+]
```
Comment (giordano) [inline P/PortMidi/build_tarballs.jl:11]: We usually prefer to build stable releases of software, and not random commits, and the `version` variable should match the upstream version number: https://docs.binarybuilder.org/stable/building/#Version-number

File: P/PortMidi/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir
+cd portmidi/
```
Comment (giordano) [inline P/PortMidi/build_tarballs.jl:16]:
```suggestion
cd $WORKSPACE/srcdir/portmidi/
```

File: P/PortMidi/build_tarballs.jl [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+cmake -DCMAKE_INSTALL_PREFIX=$prefix -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} -DCMAKE_BUILD_TYPE=Release
```
Comment (giordano) [inline P/PortMidi/build_tarballs.jl:17]:
```suggestion
cmake -DCMAKE_INSTALL_PREFIX=${prefix} \
    -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} \
    -DCMAKE_BUILD_TYPE=Release \
    -B build
```

File: P/PortMidi/build_tarballs.jl [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+make 
+make install
```
Comment (giordano) [inline P/PortMidi/build_tarballs.jl:19]:
```suggestion
cmake --build build --parallel ${nproc}
cmake --install build
```

File: P/PortMidi/build_tarballs.jl [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    Platform("armv7l", "linux"; call_abi = "eabihf", libc = "glibc"),
+    Platform("powerpc64le", "linux"; libc = "glibc"),
+    Platform("i686", "linux"; libc = "musl"),
+    Platform("x86_64", "linux"; libc = "musl"),
+    Platform("aarch64", "linux"; libc = "musl"),
+    Platform("armv6l", "linux"; call_abi = "eabihf", libc = "musl"),
+    Platform("armv7l", "linux"; call_abi = "eabihf", libc = "musl"),
+    Platform("i686", "windows"; ),
+    Platform("x86_64", "windows"; )
+]
```
Comment (giordano) [inline P/PortMidi/build_tarballs.jl:38]:
```suggestion
platforms = supported_platforms()
```

**Review by giordano — COMMENTED** [review]
Thanks for your contribution!

File: P/PortMidi/build_tarballs.jl (hunk 1 above)
Comment (SteffenPL) [reply P/PortMidi/build_tarballs.jl:11]: Thanks!

I changed it accordingly, using `v2.0.4` which is the latest stable tag release. (The software is pretty old anyway, not many new releases.)

**Comment by SteffenPL** [comment]
@giordano Thanks for all the help and corrections (and sorry for my many beginner mistakes...). Hope to add more packages soon, in particular some science stuff.

File: P/PortMidi/build_tarballs.jl (hunk 5 above)
Comment (SteffenPL) [reply P/PortMidi/build_tarballs.jl]: I added all platforms with the exception of FreeBSD, where `alsa_jll` is not supported
```julia
platforms = filter(!Sys.isfreebsd, supported_platforms()) # 'asound' build missing for freebsd
```

File: P/PortMidi/build_tarballs.jl [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+
+# The products that we will ensure are always built
+products = [
+    LibraryProduct("libportmidi", :libportmidi)
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = [
+    Dependency(PackageSpec(name="alsa_jll", uuid="45378030-f8ea-5b20-a7c7-1a9d95efb90e"), platforms=filter(!Sys.iswindows, platforms))
```
Comment (giordano) [inline P/PortMidi/build_tarballs.jl:36]: Alsa is missing also for macOS, so at this point we can just install it for Linux
```suggestion
    Dependency(PackageSpec(name="alsa_jll", uuid="45378030-f8ea-5b20-a7c7-1a9d95efb90e"), platforms=filter(Sys.islinux, platforms))
```

File: P/PortMidi/build_tarballs.jl (hunk 5 above)
Comment (giordano) [reply P/PortMidi/build_tarballs.jl]: Filtering like that is fine, you don't have a long list of platforms and it's clearer what's being excluded

**Pull request merged**